### PR TITLE
47 check comparison operators execution time

### PR DESCRIPTION
**Pull request by mla2001**

### CODE REVIEW

File: src/vtlengine/Operators/HROperators.py [hunk 1]
```diff
@@ -32,31 +32,41 @@ def imbalance_func(cls, x: Any, y: Any) -> Any:
...
+        elif hr_mode == "non_zero":
+            mask_remove = (left_series == 0) & (right_series == 0)
+            result[mask_remove] = "REMOVE_VALUE"
 
-        return func(x, y)
+        return result
 
     @classmethod
     def apply_hr_func(cls, left_series: Any, right_series: Any, hr_mode: str, func: Any) -> Any:
-        return left_series.combine(right_series, lambda x, y: cls.hr_func(x, y, hr_mode, func))
```
[the remark this answers is not in the document]
Comment (mla2001) [reply src/vtlengine/Operators/HROperators.py:56]: Added comments explaining the refactor.

File: src/vtlengine/Operators/HROperators.py [hunk 2]
```diff
@@ -75,22 +85,19 @@ def validate(cls, left_operand: Dataset, right_operand: DataComponent, hr_mode:
...
             result.data["imbalance"] = cls.apply_hr_func(
                 left.data[measure_name], right.data, hr_mode, cls.imbalance_func
             )
-        # Removing datapoints that should not be returned
-        # (we do it below imbalance calculation
-        # to avoid errors on different shape)
+
         result.data = result.data[result.data["bool_var"] != "REMOVE_VALUE"]
         result.data.drop(measure_name, axis=1, inplace=True)
         return result
```
[the remark this answers is not in the document]
Comment (mla2001) [reply src/vtlengine/Operators/HROperators.py:103]: Well, by the moment apply_hr_func is only in charge of calculating and validating the results, no about dropping them, that's why those lines are in the evaluate, however there should be no problem in moving them to the apply_hr_func.

File: src/vtlengine/Operators/HROperators.py [hunk 3]
```diff
@@ -75,22 +85,19 @@ def validate(cls, left_operand: Dataset, right_operand: DataComponent, hr_mode:
...
             result.data["bool_var"] = cls.apply_hr_func(
                 left.data[measure_name], right.data, hr_mode, cls.op_func
             )
             result.data["imbalance"] = cls.apply_hr_func(
                 left.data[measure_name], right.data, hr_mode, cls.imbalance_func
             )
-        # Removing datapoints that should not be returned
-        # (we do it below imbalance calculation
-        # to avoid errors on different shape)
+
```
[the remark this answers is not in the document]
Comment (mla2001) [reply src/vtlengine/Operators/HROperators.py:100]: Comments will be re-added.

File: src/vtlengine/Operators/Comparison.py [hunk 4]
```diff
@@ -96,19 +96,39 @@ def _cast_values(
...
             return None
         x, y = cls._cast_values(x, y)
         return cls.py_op(x, y)
 
+    import pandas as pd
+
     @classmethod
     def apply_operation_series_scalar(cls, series: Any, scalar: Any, series_left: bool) -> Any:
-        if scalar is None:
+        if scalar is None or pd.isnull(scalar):
```
Comment (javihern98) [inline src/vtlengine/Operators/Comparison.py:109]: Use only pd.isnull(scalar) here, if necessary

**Review by javihern98 — COMMENTED** [review]
Slight changes needed

File: src/vtlengine/Operators/Comparison.py (hunk 4 above)
Comment (mla2001) [reply src/vtlengine/Operators/Comparison.py:109]: Right, changes done!

File: src/vtlengine/Operators/Comparison.py [hunk 5]
```diff
@@ -96,19 +96,39 @@ def _cast_values(
...
+                elif isinstance(first_non_null, (int, float)):
+                    series = series.astype(float)
+
+        op = cls.py_op if cls.py_op is not None else cls.op_func
         if series_left:
-            return series.map(lambda x: cls.op_func(x, scalar), na_action="ignore")
+            result = series.apply(lambda x: op(x, scalar) if not pd.isnull(x) else None)
         else:
-            return series.map(lambda x: cls.op_func(scalar, x), na_action="ignore")
+            result = series.apply(lambda x: op(scalar, x) if not pd.isnull(x) else None)
```
[the remark this answers is not in the document]
Comment (mla2001) [reply src/vtlengine/Operators/Comparison.py:129]: Right, changes done!

File: src/vtlengine/Operators/Comparison.py [hunk 6]
```diff
@@ -96,19 +96,39 @@ def _cast_values(
...
+            if series_type != first_non_null_type:
+                if isinstance(first_non_null, str):
+                    series = series.astype(str)
+                elif isinstance(first_non_null, (int, float)):
+                    series = series.astype(float)
+
+        op = cls.py_op if cls.py_op is not None else cls.op_func
         if series_left:
-            return series.map(lambda x: cls.op_func(x, scalar), na_action="ignore")
+            result = series.apply(lambda x: op(x, scalar) if not pd.isnull(x) else None)
```
[the remark this answers is not in the document]
Comment (mla2001) [reply src/vtlengine/Operators/Comparison.py:127]: Right, changes done!

**Review by javihern98 — APPROVED** [review]
All good 😊

**Pull request merged**